src/cell_ref.rs

```diff
@@ -20,6 +22,22 @@ impl<'a, T> CellRef<'a, T>
 where
     T: ?Sized,
 {
+    /// Returns a clone of this `CellRef`.
+    ///
+    /// This should be used instead of `.clone()` to handle potential overflow
+    /// of references.
+    pub fn try_clone(&self) -> Result<Self, RefOverflow> {
+        let previous_value = self.flag.fetch_add(1, Ordering::Release);
```

Comment:
I'm sorry but overflow still occurs here.

Reply:
how so? the control flow returns err, so it should be detected right

Reply:
Control flow returns err, and counter is `0`. Now I can borrow thing mutably.